### ISSUE TITLE
When editing settings, add new items alphabetically sorted and fix find experience

### ISSUE THREAD

**Issue by nojvek** [Microsoft/vscode#43670]
![image](https://user-images.githubusercontent.com/1018196/36212365-97bfb70c-1158-11e8-8e6c-4da616608407.png)

If I click the settings icon and change something. That properly ends up at the end of settings.json file

I have to manually sort it by property. Once you have more than 10 settings, which is very easy, without sorting keys, it's painful to find anything. 

Add this to the fact that ctrl+f is hacked by global settings search so you can't find something inside settings.json

1) If I do ctrl+f or cmd+f inside settings.json I want the editor find experience not the fancy search experience

![image](https://user-images.githubusercontent.com/1018196/36212574-1818d9ba-1159-11e8-89e4-d2f5f9551f82.png)

2) If I change a setting by clicking the small pencil, I want that setting to be added in settings.json in alphabetically sorted position, not at the end.

`yarn add` / `npm install --save` nicely updates package.json always alphabetically sorted for dependencies. Imagine having to do that manually.

**Comment by rafaelramalho19** [comment]
I would like to start contributing to VSCode, if it's fine by you guys can I pick this issue up? :)

**Comment by roblourens** [comment]
@rafaelramalho19 I don't think this is an easy first issue. There is an extension, Sort JSON Keys, which helps with this. I would want slightly different sorting behavior, grouping settings and putting empty lines between groups. An extension that does that, and updates the settings.json automatically when it changes, would be possible

For the ctrl+f binding, I use this

```json
{
    "key": "cmd+f",
    "command": "-settings.action.search",
    "when": "inSettingsEditor"
},
{
    "key": "alt+cmd+f",
    "command": "settings.action.search",
    "when": "inSettingsEditor"
}
```

which may be closer to what you want.

**Comment by nojvek** [comment]
Didn’t know about sort Json keys. Will try it.

I do feel that breaking ctrl+f when I am editing user or workspace settings
is bad UX.

Ctrl+f when focus is on default settings does the fancy smart search.

Would you agree?

**Comment by nojvek** [comment]
Here's a real scenario that I encounter everytime I edit settings:

I love vscode is due to its customizability + extensions and I have a config that's slowly grown to 100+lines. 

Suppose I want to disable python flake8 

I type `cmd+f` and then `flake8`

![image](https://user-images.githubusercontent.com/1018196/36434898-01cf0f42-1615-11e8-9945-fcf77089be44.png)

vscode shows the default settings related to flake8 but it wont find anything in user settings since its out of scroll. So I have to manually scroll with a mouse to see if it found anything.

Now I can't hit Esc and make the mouse cursor go to the nearest find. Esc clears the filter so you are back to square zero.

Essentially the smart search is super useful for default settings but not very useful for user settings. 

--

Now If I use the editor search, the experience is a lot nicer

I type `cmd+f` + `fl` and user settings has already scrolled to flake8 settings. I can press enter to select the setting I want.

![image](https://user-images.githubusercontent.com/1018196/36435196-d9d1b692-1615-11e8-885f-8a2007354f68.png)

Pressing `ESC` puts the cursor near the match I selected

![image](https://user-images.githubusercontent.com/1018196/36435246-ff2e10e8-1615-11e8-8e58-d530324220f7.png)

Now it's just a matter of `alt+shift+right` to move cursor between words and change the setting. I can be done in a second or two without using the mouse at all.

The editor search allows navigation of editor with purely my keyboard, the smart search does not.

--

So please please please please please don't make `cmd+f` / `ctrl+f` from the editor go to smart search. Please make it go to editor search. If I press `cmd+f` / `ctrl+f` in default settings, then smart search is fine. It's the most useful for default settings search

**Comment by nojvek** [comment]
```
{
    "key": "cmd+f",
    "command": "-settings.action.search",
    "when": "inSettingsEditor"
},
{
    "key": "alt+cmd+f",
    "command": "settings.action.search",
    "when": "inSettingsEditor"
}
```

This is a beautiful workaround @roblourens. Thanks

I would still like this to be fixed at default level for all users. I also noticed the default search doesn't update scroll overlay for find markers.
